Lection 3/task 3.1/ReportApp/Services/ReportGenerateService.cs

```diff
@@ -0,0 +1,57 @@
+﻿using ReportApp.Services.Activity;
+using ReportApp.Services.Shop;
+
+namespace ReportApp.Services;
+
+public class ReportGenerateService
+{
+    private readonly ConfigurationService _reportConfigurationService = new ConfigurationService();
+    private readonly TemplateManagerService _templateService = new TemplateManagerService();
+
+    private readonly ActivityTemplateManagerService _activityReportDataService = new ActivityTemplateManagerService();
+    private readonly ShopTemplateManagerService _shopReportDataService = new ShopTemplateManagerService();
+
+    private readonly ActivitySerializService _activitySerializService = new ActivitySerializService();
+    private readonly ShopSerializService _shopSerializService = new ShopSerializService();
+
+    public void GenerateReport(string pathToFile, string type)
+    {
+        Action action = type switch
+        {
+            "Activity" => () => GenerateActivityReport(pathToFile, type),
+            "Shop" => () => GenerateShopReport(pathToFile, type),
+            _ => throw new ArgumentException("It's a wrong report type"),
+        };
+
+        action();
+    }
+
+    private void GenerateActivityReport(string pathToFile, string type)
+    {
+        string pathToConfiguration = "./ReportConfigurations/Activity.json";
+
+        var configuration = _reportConfigurationService.GetActivityConfiguration(pathToConfiguration);
```

Comment:
Even in usage those methods like GetActivityConfig is making you thing there is a better way